client/components/sites-dropdown/docs/example.jsx

```diff
@@ -1,18 +1,55 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import find from 'lodash/find';
+import { urlToSlug } from 'lib/url';
+
 
 /**
  * Internal dependencies
  */
 import SitesDropdown from 'components/sites-dropdown';
 
-export default React.createClass( {
+class SitesDropdownExample extends PureComponent {
+	static propTypes = {
+		selectedSiteId: React.PropTypes.number
+	}
+
+	constructor( props ) {
+		super( props );
 
-	displayName: 'SitesDropdown',
+		this.state = {
+			selectedSiteId: this.props.selectedSiteId
+		};
+	}
+
+	updateSelectedSite = ( siteSlug ) => {
+		const selectedSite = find( this.props.sites, site => urlToSlug( site.URL ) === siteSlug );
+		if ( selectedSite ) {
+			this.setState( { selectedSiteId: selectedSite.ID } );
+		}
+	}
 
-	render: function() {
-		return <SitesDropdown />;
+	render(){
+		return (
+			<SitesDropdown
+				selectedSiteId={ this.state.selectedSiteId }
```

Comment:
It's probably better to defer to default handling in SitesDropdown than add logic in the example.